src/core_modules/capture-ui/IconButton/IconButton.component.js

```diff
@@ -15,12 +41,18 @@ export const IconButton = ({ children, className, dataTest, onClick, ...passOnPr
         event.preventDefault();
     }, []);
 
+    const onClickHandler = useCallback((event: SyntheticMouseEvent<HTMLSpanElement>) => {
+        if (disabled) return null;
+        return onClick(event);
+    }, [disabled, onClick]);
+
     return (
         <span
             {...passOnProps}
-            onClick={onClick}
+            onClick={onClickHandler}
+            disabled={disabled}
             data-test={dataTest}
```

Comment:
It seems that many Cypress tests are failing due to missing `dataTest` prop. Can you have a look?